Library/Tracking/Koala.swift

```diff
@@ -2052,45 +2042,64 @@ public final class Koala {
   }
 }
 
+// MARK: - Project Properties
+
+// FIXME: remove the function below and replace with projectProperties
+
 private func properties(
   project: Project,
   loggedInUser: User?,
   prefix: String = "project_"
+) -> [String: Any] {
+  return projectProperties(from: project, loggedInUser: loggedInUser, prefix: prefix)
+}
+
+private func projectProperties(
+  from project: Project,
+  reward _: Reward? = nil,
+  backing _: Backing? = nil,
+  loggedInUser: User? = nil,
+  dateType: DateProtocol.Type = AppEnvironment.current.dateType,
+  calendar: Calendar = AppEnvironment.current.calendar,
+  prefix: String = "project_"
 ) -> [String: Any] {
   var props: [String: Any] = [:]
 
   props["backers_count"] = project.stats.backersCount
+  props["category"] = project.category.name
   props["country"] = project.country.countryCode
+  props["comments_count"] = project.stats.commentsCount
   props["currency"] = project.country.currencyCode
+  props["creator_uid"] = project.creator.id
+  props["deadline"] = project.dates.deadline
   props["goal"] = project.stats.goal
-  props["pid"] = project.id
+  props["launched_at"] = project.dates.launchedAt
+  props["location"] = project.location.name
   props["name"] = project.name
-  props["pledged"] = project.stats.pledged
+  props["pid"] = project.id
+  props["parent_category"] = project.category.parent?.name
   props["percent_raised"] = project.stats.fundingProgress
-  props["has_video"] = project.video != nil
+  props["pledged"] = project.stats.pledged
   props["state"] = project.state.rawValue
-  props["update_count"] = project.stats.updatesCount
-  props["comments_count"] = project.stats.commentsCount
-
-  let now = AppEnvironment.current.dateType.init().timeIntervalSince1970
-  props["hours_remaining"] = Int(ceil(max(0.0, (project.dates.deadline - now) / 3_600.0)))
-  props["duration"] = Int(round(project.dates.deadline - project.dates.launchedAt))
+  props["static_usd_rate"] = project.stats.staticUsdRate
+  props["current_pledge_amount_usd"] = project.stats.pledgedUsd
+  props["goal_usd"] = project.stats.goalUsd
+  props["has_video"] = project.video != nil
 
-  props["category"] = project.category.name
-  props["parent_category"] = project.category.parent?.name
+  let now = dateType.init().date
+  props["hours_remaining"] = project.dates.hoursRemaining(from: now, using: calendar)
```

Comment:
For projects that have ended in the past this tracks a negative value. Do we want to `min` that as `0`?

Reply:
Good idea, nice catch.